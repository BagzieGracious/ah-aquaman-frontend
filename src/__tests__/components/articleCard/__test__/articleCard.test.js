import React from "react";
import { shallow } from "enzyme";
import ArticleCard from "../../../../components/article/articleCard";

describe("ArticleCard component test", () => {
  const props = {
    author: {
      username: "testuser",
    },
    title: "test Article",
    description: "this is test description",
    body: "This is a test body",
  };
  const wrapper = shallow(<ArticleCard {...props} />);

  it("Should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
