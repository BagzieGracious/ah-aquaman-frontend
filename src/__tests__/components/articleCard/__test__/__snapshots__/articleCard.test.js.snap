// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard component test Should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleCard
    author={
      Object {
        "username": "testuser",
      }
    }
    body="This is a test body"
    description="this is test description"
    title="test Article"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="article-img"
        >
          <img
            alt=""
            className="img-fluid"
          />
        </div>,
        <div
          className="article-text"
        >
          <div
            className="inner-text"
          >
            <Link
              className="article-title"
              to="/article/undefined"
            >
              <h4>
                test Article
              </h4>
            </Link>
            <p>
              This is a test body
            </p>
            <div
              className="date"
            >
              <p>
                <i
                  className="fa fa-calendar"
                />
                 
                Invalid Date
                     
                <i
                  className="fa fa-comments-o"
                />
                 
                24
              </p>
            </div>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            className="img-fluid"
          />,
          "className": "article-img",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "img-fluid",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="inner-text"
          >
            <Link
              className="article-title"
              to="/article/undefined"
            >
              <h4>
                test Article
              </h4>
            </Link>
            <p>
              This is a test body
            </p>
            <div
              className="date"
            >
              <p>
                <i
                  className="fa fa-calendar"
                />
                 
                Invalid Date
                     
                <i
                  className="fa fa-comments-o"
                />
                 
                24
              </p>
            </div>
          </div>,
          "className": "article-text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="article-title"
                to="/article/undefined"
              >
                <h4>
                  test Article
                </h4>
              </Link>,
              <p>
                This is a test body
              </p>,
              <div
                className="date"
              >
                <p>
                  <i
                    className="fa fa-calendar"
                  />
                   
                  Invalid Date
                       
                  <i
                    className="fa fa-comments-o"
                  />
                   
                  24
                </p>
              </div>,
            ],
            "className": "inner-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h4>
                  test Article
                </h4>,
                "className": "article-title",
                "to": "/article/undefined",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test Article",
                },
                "ref": null,
                "rendered": "test Article",
                "type": "h4",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This is a test body",
              },
              "ref": null,
              "rendered": "This is a test body",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  <i
                    className="fa fa-calendar"
                  />
                   
                  Invalid Date
                       
                  <i
                    className="fa fa-comments-o"
                  />
                   
                  24
                </p>,
                "className": "date",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="fa fa-calendar"
                    />,
                    " ",
                    "Invalid Date",
                    "     ",
                    <i
                      className="fa fa-comments-o"
                    />,
                    " ",
                    24,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-calendar",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " ",
                  "Invalid Date",
                  "     ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-comments-o",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " ",
                  24,
                ],
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="article-img"
          >
            <img
              alt=""
              className="img-fluid"
            />
          </div>,
          <div
            className="article-text"
          >
            <div
              className="inner-text"
            >
              <Link
                className="article-title"
                to="/article/undefined"
              >
                <h4>
                  test Article
                </h4>
              </Link>
              <p>
                This is a test body
              </p>
              <div
                className="date"
              >
                <p>
                  <i
                    className="fa fa-calendar"
                  />
                   
                  Invalid Date
                       
                  <i
                    className="fa fa-comments-o"
                  />
                   
                  24
                </p>
              </div>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              className="img-fluid"
            />,
            "className": "article-img",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "img-fluid",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="inner-text"
            >
              <Link
                className="article-title"
                to="/article/undefined"
              >
                <h4>
                  test Article
                </h4>
              </Link>
              <p>
                This is a test body
              </p>
              <div
                className="date"
              >
                <p>
                  <i
                    className="fa fa-calendar"
                  />
                   
                  Invalid Date
                       
                  <i
                    className="fa fa-comments-o"
                  />
                   
                  24
                </p>
              </div>
            </div>,
            "className": "article-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="article-title"
                  to="/article/undefined"
                >
                  <h4>
                    test Article
                  </h4>
                </Link>,
                <p>
                  This is a test body
                </p>,
                <div
                  className="date"
                >
                  <p>
                    <i
                      className="fa fa-calendar"
                    />
                     
                    Invalid Date
                         
                    <i
                      className="fa fa-comments-o"
                    />
                     
                    24
                  </p>
                </div>,
              ],
              "className": "inner-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <h4>
                    test Article
                  </h4>,
                  "className": "article-title",
                  "to": "/article/undefined",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "test Article",
                  },
                  "ref": null,
                  "rendered": "test Article",
                  "type": "h4",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This is a test body",
                },
                "ref": null,
                "rendered": "This is a test body",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    <i
                      className="fa fa-calendar"
                    />
                     
                    Invalid Date
                         
                    <i
                      className="fa fa-comments-o"
                    />
                     
                    24
                  </p>,
                  "className": "date",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fa fa-calendar"
                      />,
                      " ",
                      "Invalid Date",
                      "     ",
                      <i
                        className="fa fa-comments-o"
                      />,
                      " ",
                      24,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-calendar",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " ",
                    "Invalid Date",
                    "     ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-comments-o",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " ",
                    24,
                  ],
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
